pkg/kmodule/kmodule_linux.go

```diff
@@ -19,57 +20,57 @@ const (
 )
 
 // Init loads the kernel module given by image with the given options.
-func Init(image []byte, opts string) error {
+func Init(image []byte, opts string) (error, syscall.Errno) {
 	optsNull, err := unix.BytePtrFromString(opts)
 	if err != nil {
-		return fmt.Errorf("kmodule.Init: could not convert %q to C string: %v", opts, err)
+		return fmt.Errorf("kmodule.Init: could not convert %q to C string: %v", opts, err), 0
 	}
 
 	if _, _, e := unix.Syscall(unix.SYS_INIT_MODULE, uintptr(unsafe.Pointer(&image[0])), uintptr(len(image)), uintptr(unsafe.Pointer(optsNull))); e != 0 {
-		return fmt.Errorf("init_module(%v, %q) failed with %v", image, opts, e)
+		return fmt.Errorf("init_module(%v, %q) failed with %v", image, opts, e), e
 	}
 
-	return nil
+	return nil, 0
 }
 
 // FileInit loads the kernel module contained by `f` with the given opts and
 // flags.
 //
 // FileInit falls back to Init when the finit_module(2) syscall is not available.
-func FileInit(f *os.File, opts string, flags uintptr) error {
+func FileInit(f *os.File, opts string, flags uintptr) (error, syscall.Errno) {
 	optsNull, err := unix.BytePtrFromString(opts)
 	if err != nil {
-		return fmt.Errorf("kmodule.Init: could not convert %q to C string: %v", opts, err)
+		return fmt.Errorf("kmodule.Init: could not convert %q to C string: %v", opts, err), 0
 	}
 
 	if _, _, e := unix.Syscall(unix.SYS_FINIT_MODULE, f.Fd(), uintptr(unsafe.Pointer(optsNull)), flags); e == unix.ENOSYS {
 		if flags != 0 {
-			return fmt.Errorf("finit_module unavailable")
+			return fmt.Errorf("finit_module unavailable"), e
 		}
 
 		// Fall back to regular init_module(2).
 		img, err := ioutil.ReadAll(f)
 		if err != nil {
-			return fmt.Errorf("kmodule.FileInit: %v", err)
+			return fmt.Errorf("kmodule.FileInit: %v", err), 0
 		}
 		return Init(img, opts)
 	} else if e != 0 {
-		return fmt.Errorf("finit_module(%v, %q, %#x) failed with %v", f, opts, flags, e)
+		return fmt.Errorf("finit_module(%v, %q, %#x) failed with %v", f, opts, flags, e), e
 	}
 
-	return nil
+	return nil, 0
 }
 
 // Delete removes a kernel module.
-func Delete(name string, flags uintptr) error {
+func Delete(name string, flags uintptr) (error, syscall.Errno) {
```

Comment:
This is a hack. You shouldn't return two errors. If you need the errno, use something like this:

```go
type SyscallError struct {
  Msg string
  Errno syscall.Errno
}

func (s SyscallError) Error() string {
  return fmt.Sprintf("%s: %v", s.Msg, s.Errno)
}
```

Reply:
Also, use `unix.Errno`.

Reply:
Oh yes, that makes sense.

Reply:
@hugelgupf I think the type is `syscall.Errno` (that is also the type returned by the syscall API). There is no `unix.Errno`. These are just consts that are casted to `syscall.Errno`. So I kept the `Errno` field type of `SyscallError` as `syscall.Errno`. OK?

Reply:
Ah, okay, sorry. I blindly assumed the unix package had everything the syscall package did for unix.